In [1]:
from main import *

<span style="font-size: 30px;">Few waves example for y=sin(w1⋅t)+sin(w2⋅t)</span>

<img src="plots/rep08-double-better/double_sine_linear_mu0.0_sigma0.1_w11.30_w29.57.png" width="420"> <img src="plots/rep08-double-better/double_sine_linear_mu0.0_sigma0.1_w16.55_w29.12.png" width="420">


<span style="font-size: 30px;">Few waves example for y=sin(w1⋅t)+sin(w2⋅t)+sin(w1⋅t) * sin(w2⋅t)</span>

<img src="plots/rep08-double-better/double_sine_product_mu0.0_sigma0.1_w11.30_w29.57.png" width="420"> <img src="plots/rep08-double-better/double_sine_product_mu0.0_sigma0.1_w19.38_w25.79.png" width="420">



<span style="font-size: 30px;">Contour plots</span>

- I added contour plots (analytical view - nothing from model used) for both signals. 
- i plotted both options with just few discrete layers and with smooth transitions between layers (coloured) 

<img src="plots/rep08-double-better/analytic_contour_fc_linear_t2.00_triple_dt1.0_w0.5-10.png" width="1000">
<img src="plots/rep08-double-better/analytic_contour_fc_product_t2.00_triple_dt1.0_w0.5-10.png" width="1000">


<img src="plots/rep08-double-better/analytic_contour_nc_linear_t2.00_triple_dt1.0_w0.5-10.png" width="1000">
<img src="plots/rep08-double-better/analytic_contour_nc_product_t2.00_triple_dt1.0_w0.5-10.png" width="1000">


In [2]:
#y=sin(w1⋅t) + sin(w2⋅t) - w1,w2 from [0.5,10.0]
main3(dataset="linear")

Test MSE 0.015823, MAE 0.097533


<span style="font-size: 22px;">Results for y=sin(w1⋅t) + sin(w2⋅t) - w1,w2 from [0.5,10.0]</span>

<img src="plots/rep08-double-better/T3_linear_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.1_PREDvsREAL2.png" width="900">

In [3]:
#y=sin(w1⋅t) + sin(w2⋅t) + sin(w1⋅t)*sin(w2⋅t) - w1,w2 from [0.5,10.0]
main3(dataset="nonlinear")

Test MSE 0.013789, MAE 0.087584


<span style="font-size: 22px;">Results for y=sin(w1⋅t) + sin(w2⋅t) + sin(w1⋅t)*sin(w2⋅t) - w1,w2 from [0.5,10.0]</span>

- Transformer with a flow-only probabilistic head reliably recovers frequencies in both linear and nonlinear mixtures

<img src="plots/rep08-double-better/T3_nonlinear_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.1_PREDvsREAL2.png" width="900">

Extensive tuning of the flow does not significantly reduce the gap between predicted mean and true frequencies for N=1000.
Increasing the dataset size improves accuraccy, but does not move mean closer to true frequency, but signal length, noise level, and discretization could help (not sure about reducing gap between mean and true value). Maybe combining likelihood with additional constraints could help, but I am not sure for purely likelihood-based probabilistic model.

<span style="font-size: 30px;">When increased sigma noise</span>

N=1000, w₁,w∈[0.5,10], t∈[0,2π],

- std (σ)= 0.1 Val MSE: **0.026**470, Test MSE 0.015823, MAE 0.097533
- std (σ)= 0.2 Val MSE: **0.014**076, Test MSE 0.017569, MAE 0.097023
- std (σ)= 0.3 Val MSE: **0.025**700, Test MSE 0.029101, MAE 0.126703
- std (σ)= 0.4 Val MSE: **0.036**818, Test MSE 0.043599, MAE 0.148789
- std (σ)= 0.5 Val MSE: **0.067**085, Test MSE 0.068242, MAE 0.192128
- std (σ)= 0.6 Val MSE: **0.073**738, Test MSE 0.089586, MAE 0.218073
- std (σ)= 0.7 Val MSE: **0.147**853, Test MSE 0.140084, MAE 0.258257
- std (σ)= 0.8 Val MSE: **0.402**019, Test MSE 0.378684, MAE 0.377969
- std (σ)= 0.9 Val MSE: **0.501**077, Test MSE 0.602784, MAE 0.426835
- std (σ)= 1.0 Val MSE: **1.008**297, Test MSE 0.684471, MAE 0.485811
- std (σ)=1.25 Val MSE: **2.077**646, Test MSE 1.552667, MAE 0.804854

As the noise standard deviation increases, both validation and test errors grow monotonically, indicating a gradual degradation of frequency identifiability.
For small noise levels < 0.3, the model remains relatively stable, with only a mild increase in MAE - as was expected.


<img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.1_PREDvsREAL2.png" width="550"> <img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.2_PREDvsREAL2.png" width="550">

<img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.3_PREDvsREAL2.png" width="550"> <img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.4_PREDvsREAL2.png" width="550">

<img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.5_PREDvsREAL2.png" width="550"> <img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.6_PREDvsREAL2.png" width="550">

<img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.7_PREDvsREAL2.png" width="550"> <img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.8_PREDvsREAL2.png" width="550">

<img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std0.9_PREDvsREAL2.png" width="550"> <img src="plots/rep08-double-better/T3_double_sorted_N1000_tdis100_w0.5-10_seed177013_std1.0_PREDvsREAL2.png" width="550">